scripts/TransferIndexConfig.js

```diff
@@ -43,25 +44,35 @@ class TransferIndexConfigScript extends Base {
     this.destinationIndex = destinationClient.initIndex(options.indexName);
   }
 
-  async transferIndexConfig() {
-    // Transfer settings, synonyms, and query rules
-    const settings = await this.sourceIndex.getSettings();
-    const synonyms = await this.sourceIndex.exportSynonyms();
-    const rules = await this.sourceIndex.exportRules();
-    const sOptions = {
-      forwardToReplicas: true,
-      replaceExistingSynonyms: true,
-    };
-    const rOptions = {
-      forwardToReplicas: true,
-      clearExistingRules: true,
-    };
-    await this.destinationIndex.setSettings(settings);
-    await this.destinationIndex.batchSynonyms(synonyms, sOptions);
-    await this.destinationIndex.batchRules(rules, rOptions);
+  setConfigOptions(options) {
+    this.sOptions = {};
+    this.rOptions = {};
+    if (!options.configParams) return;
+    const configParams = JSON.parse(options.configParams);
+    if (configParams.batchSynonymsParams)
+      this.sOptions = configParams.batchSynonymsParams;
+    if (configParams.batchRulesParams)
+      this.rOptions = configParams.batchRulesParams;
   }
 
-  start(program) {
+  transferIndexConfig() {
+    return new Promise(async (resolve, reject) => {
+      try {
+        // Transfer settings, synonyms, and query rules
+        const settings = await this.sourceIndex.getSettings();
+        const synonyms = await this.sourceIndex.exportSynonyms();
+        const rules = await this.sourceIndex.exportRules();
+        await this.destinationIndex.setSettings(settings);
+        await this.destinationIndex.batchSynonyms(synonyms, this.sOptions);
+        await this.destinationIndex.batchRules(rules, this.rOptions);
+        resolve();
```

Comment:
can you return as well ? 
```
try {
return resolve(e);
} catch(e){
 return reject(e);
}
```
otherwise the thread won't stop
```
try {
resolve(e);
console.log('toto');
} catch(e){
 return reject(e);
}
```
will print 'toto'